### ISSUE TITLE
kubectl explain application gives no content

### ISSUE THREAD

**Issue by r0bj** [argoproj/argo-cd#5840]
Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

kubectl explain application gives no content:
```
$ kubectl explain application
KIND:     Application
VERSION:  argoproj.io/v1alpha1

DESCRIPTION:
     <empty>
```

Usually there is description for CRD and its nested fields, e.g.:
```
$ kubectl explain prometheuses
KIND:     Prometheus
VERSION:  monitoring.coreos.com/v1

DESCRIPTION:
     Prometheus defines a Prometheus deployment.

FIELDS:
   apiVersion	<string>
     APIVersion defines the versioned schema of this representation of an
     object. Servers should convert recognized schemas to the latest internal
     value, and may reject unrecognized values. More info:
     https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources

   kind	<string>
     Kind is a string value representing the REST resource this object
     represents. Servers may infer this from the endpoint the client submits
     requests to. Cannot be updated. In CamelCase. More info:
     https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds

   metadata	<Object>
     Standard object's metadata. More info:
     https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata

   spec	<Object> -required-
     Specification of the desired behavior of the Prometheus cluster. More info:
     https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#spec-and-status

   status	<Object>
     Most recent observed status of the Prometheus cluster. Read-only. Not
     included when requesting from the apiserver, only from the Prometheus
     Operator API itself. More info:
     https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#spec-and-status
```

**To Reproduce**

1. Install ArgoCD: kubectl apply -n argocd -f https://raw.githubusercontent.com/argoproj/argo-cd/stable/manifests/install.yaml
2. Check CRD explain: kubectl explain application

**Expected behavior**

`kubectl explain application` gives description of `Application` CRD and all its nested fields.

**Screenshots**

**Version**

```shell
argocd: v1.8.7+eb3d1fb.dirty
  BuildDate: 2021-03-07T20:25:24Z
  GitCommit: eb3d1fb84b9b77cdffd70b14c4f949f1c64a9416
  GitTreeState: dirty
  GoVersion: go1.16
  Compiler: gc
  Platform: darwin/amd64
argocd-server: v1.8.7+eb3d1fb
  BuildDate: 2021-03-03T07:13:34Z
  GitCommit: eb3d1fb84b9b77cdffd70b14c4f949f1c64a9416
  GitTreeState: clean
  GoVersion: go1.14.12
  Compiler: gc
  Platform: linux/amd64
  Ksonnet Version: v0.13.1
  Kustomize Version: v3.8.1 2020-07-16T00:58:46Z
  Helm Version: v3.4.1+gc4e7485
  Kubectl Version: v1.17.8
  Jsonnet Version: v0.17.0
```

**Comment by kshamajain99** [comment]
It works for me
```
$ kubectl explain Application
KIND:     Application
VERSION:  argoproj.io/v1alpha1

DESCRIPTION:
     Application is a definition of Application resource.

FIELDS:
   apiVersion	<string>
     APIVersion defines the versioned schema of this representation of an
     object. Servers should convert recognized schemas to the latest internal
     value, and may reject unrecognized values. More info:
     https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources

   kind	<string>
     Kind is a string value representing the REST resource this object
     represents. Servers may infer this from the endpoint the client submits
     requests to. Cannot be updated. In CamelCase. More info:
     https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds

   metadata	<Object> -required-
     Standard object's metadata. More info:
     https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata

   operation	<Object>
     Operation contains information about a requested or running operation

   spec	<Object> -required-
     ApplicationSpec represents desired application state. Contains link to
     repository with application definition and additional parameters link
     definition revision.

   status	<Object>
     ApplicationStatus contains status information for the application
```

**Comment by r0bj** [comment]
I've just created fresh cluster:
1. kind create cluster
2. kubectl create ns argocd
3. kubectl apply -n argocd -f https://raw.githubusercontent.com/argoproj/argo-cd/stable/manifests/install.yaml
4. kubectl explain application
```
KIND:     Application
VERSION:  argoproj.io/v1alpha1

DESCRIPTION:
     <empty>
```

**Comment by jannfis** [comment]
This might be due to the version of `apiextensions.k8s.io` that differs between `stable` (which is currently pointing to `v1.8.7`) and latest master code. We migrated the CRDs from `apiextensions.k8s.io/v1alpha1` to `apiextensions.k8s.io/v1` just recently.

I think `kubectl explain` does not work with CRDs created in `apiextensions.k8s.io/v1alpha1`

**Comment by jannfis** [comment]
Closing this issue as it's been fixed in 2.0
